### ISSUE TITLE
Add (optional) disk usage information on `trash-list`

### ISSUE THREAD

**Issue by eloyesp** [andreafrancia/trash-cli#149]
It would be awesome to add disk usage information on trash list to help when disk is getting full what the problematic file could be.

**Comment by UnicodeAlt255** [comment]
@eloyesp Why not just use `ncdu`?

**Comment by antofthy** [comment]
@billop
1/ not everyone has ncdu (non-standard)
2/ It is not even that good (can't read normal du output from a remote system)
3/ it's not very user friendly with regards to the desktop trash which uses randomly generated filenames (to protect again unusual filenames), and keeps trashed directories together as complete unit.

That last is also why it is a BAD idea to use 'find' to delete older files in the desktop trash.
Basically you need more smarts than normal system utilities, and why I am upvoting this request myself.

**Comment by eloyesp** [comment]
@billop I didn't know about `ncdu`, I've been using that now and it's great.

On the other hand, adding to what @antofthy mentions, I'm not sure what would happen with the corresponding `.trashinfo` files when I remove files manually.

**Comment by antofthy** [comment]
The problem is find will remove old files that were already in the directories just trashed.
That is you trash a directory with some old files it in.  Next 'find' run and those old files get removed, even though the directory was just trashed!   Basically Trashcan is designed to not only keep a backup of files, but keep a backup with there original file meta data, including age!

Find is a bad idea...

From the trashcan spec...
[](https://specifications.freedesktop.org/trash-spec/trashspec-latest.html)

> If a directory was trashed in its entirety, it is easiest to undelete it or remove it from the trash only in its entirety as well, not as separate files.

**Comment by billop** [comment]
> For DU alternatives (many better than "ncdu") see
> http://www.ict.griffith.edu.au/anthony/info/apps/du.txt

So I looked at that list but it seems like ncdu is the only notable ncurses tool. Unless you were talking about the GUI alternatives, which I personally like filelight for but I personally prefer ncdu these days.

**Comment by antofthy** [comment]
My main problem with just about EVERY du graphing utility, is they don't accept standard 'du' output!
I often do a du into a text file, transfer it to my workstation so I can then use a graphical du viewer.
ncdu is pretty good, but I don't want to try to install it on 800+ machines of all sorts of OS's and versions, where 'du' is on all of them!

**Comment by antofthy** [comment]
Why is it even 'hidden' then?   Especially a it is a very useful option!

**Comment by andreafrancia** [comment]
> Why is it even 'hidden' then? Especially a it is a very useful option!

Because I found the time to implement it and I don't have found the time to document it. Do a pull request if you care enough.

**Comment by andreafrancia** [comment]
And the other reason is that I need some time to decide if it is a good idea to keep it and to keep it implemented like at it is implemented.
